cmd/snap/cmd_prepare_image.go

```diff
@@ -29,6 +30,9 @@ import (
 )
 
 type cmdPrepareImage struct {
+	Classic             bool   `long:"classic"`
+	ClassicArchitecture string `long:"classic-arch"`
```

Comment:
I think this is a poor choice. I'd either have it be `--architecture` (or `--arch`), and error out if it's different from the one in the model (i.e. not block using it on core), or make it be an optional argument of the `--classic` option.

For the latter, something like
```
Classic string `long:"classic" optional:"true" optional-value:"from-model"
```
should do the trick.